thrust/thrust/device_ptr.h

```diff
@@ -154,6 +158,13 @@ class device_ptr
     return *this;
   }
 
+#if _CCCL_HAS_CTK()
+  //! \brief Converts this \c device_ptr to a raw \c CUdeviceptr.
+  _CCCL_HOST_DEVICE operator CUdeviceptr() const noexcept
```

Comment:
Critical: Unfortunately, this breaks a lot of tests with gcc, because now `device_ptr + offset` is ambigious, for example:
```
/home/bgruber/dev/cccl/thrust/testing/cuda/device_ptr.cu:8:60:   required from here
    8 |   thrust::device_vector<int> vec(3);
      |                                                            ^
/home/bgruber/dev/cccl/lib/cmake/thrust/../../../thrust/thrust/system/cuda/detail/uninitialized_fill.h:79:14: error: ISO C++ says that these are ambiguous, even though the worst conversion for the first is better than the worst conversion for the second: [-Werror]
   79 |   return first + count;
      |        ~~~~~~^~~~~~~
```
The operator needs to be `explicit`.